lambda/check_lambda_modules.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# © 2024 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+# stop on error
+set -e
+
+pwd="$PWD"
+base_dir="$1"
+cd "$base_dir"
+echo "Checking installed lambda files and modules"
+
+if [ ! -f "python/otel_wrapper.py" ]; then
+    echo "FAILED: Missing lambda otel_wrapper"
+    exit 1
+fi
+
+if [ ! -f "solarwinds-apm/wrapper" ]; then
+    echo "FAILED: Missing SolarWinds lambda wrapper"
+    exit 1
+fi
+
+expected_sw_extension_files="./python/solarwinds_apm/extension/VERSION
+./python/solarwinds_apm/extension/__init__.py
+./python/solarwinds_apm/extension/_oboe.*.so
+./python/solarwinds_apm/extension/bson
+./python/solarwinds_apm/extension/bson/bson.h
+./python/solarwinds_apm/extension/bson/platform_hacks.h
+./python/solarwinds_apm/extension/liboboe.so
+./python/solarwinds_apm/extension/oboe.py"
+found_sw_extension_files=$(find ./python/solarwinds_apm/extension -not -path ./python/solarwinds_apm/extension -a -not -name '*.pyc' -a -not -name '__pycache__' | LC_ALL=C sort)
```

Comment:
Minor, it seems in the make recipe you're already removing `__pycache__`, wondering if the not *.pyc and __pycache__ are still needed?

Reply:
The build _should_ be removing them but looks like these are included! 🤦‍♀️ 

```
./python/solarwinds_apm/extension/__pycache__
./python/solarwinds_apm/extension/__pycache__/__init__.cpython-38.pyc
./python/solarwinds_apm/extension/__pycache__/oboe.cpython-38.pyc
```

I'll merge this PR, and work on another PR to fix the build and check.

Reply:
Actually wait. It's this step that's happening after the check that removes them before archiving:

`@find ${target_dir}/python -type d -name '__pycache__' | xargs rm -rf`

so I'll reorder things

Reply:
Addressed in [02b205a](https://github.com/solarwinds/apm-python/pull/346/commits/02b205a8d4fa50e010c5b8a04d8c86d9ef92eaaf)

Update check (and build) ran on `custom` branch successfully: https://github.com/solarwinds/apm-python/actions/runs/8471848476/job/23212703110